jetty-core/jetty-tests/jetty-test-client-transports/src/test/java/org/eclipse/jetty/test/client/transport/HttpClientTest.java

```diff
@@ -1024,9 +1026,19 @@ public void testRequestConnection(Transport transport) throws Exception
         ContentResponse response = client.newRequest(newURI(transport))
             .onRequestBegin(r ->
             {
-                if (r.getConnection() == null)
+                Connection connection = r.getConnection();
+                if (connection == null)
                     r.abort(new IllegalStateException());
             })
+            .onRequestHeaders(r ->
+            {
+                if (transport.isSecure())
+                {
+                    EndPoint.SslSessionData sslSessionData = r.getConnection().getSslSessionData();
+                    if (sslSessionData == null)
+                        r.abort(new IllegalStateException());
```

Comment:
So we are saying that having a non null `SslSessionData`, even with all the fields null, is saying that the endpoint/connection is Ssl.  I think this is consistent with other usage in the code.

Reply:
Yes, it's just that we could not gather the TLS data (cipher, certificates, etc.).